packages/server/modules/serverinvites/domain/operations.ts

```diff
@@ -0,0 +1,114 @@
+import { ServerRoles, StreamRoles } from '@speckle/shared'
+import { UserWithOptionalRole } from '@/modules/core/repositories/users'
+import { ResourceTargets } from '@/modules/serverinvites/helpers/inviteHelper'
+import {
+  ServerInviteRecord,
+  StreamInviteRecord
+} from '@/modules/serverinvites/domain/types'
+import { StreamWithOptionalRole } from '@/modules/core/repositories/streams'
+
+export type QueryAllUserStreamInvites = (
+  userId: string
+) => Promise<StreamInviteRecord[]>
+
+type FindStreamInviteArgs = {
```

Comment:
The type doesn't have to be defined by the first implementation, but it can live next to it since they're extremely closely related, and as I mentioned before - 99% of the time there is not a second implementation.

Feels like we're valuing some abstract theoretical concern higher than actual development experience. Ultimately what matters is developer experience and maintainability, I mean that is the purpose why sometimes you want to keep the types separate from implementation - cause if there are multiple, it doesn't make sense for the type to have affinity for one over the other. But in this case, with 1 implementation virtually every time, this is not a real issue.